js/data.js

```diff
@@ -6,3 +6,18 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+
+var form = document.querySelector('.form');
+
+form.addEventListener('reset', beforeReset);
+function beforeReset(event) {
+  var dataJSON = JSON.stringify(data);
+  localStorage.setItem('data', dataJSON);
+}
+
+var previousDataJSON = localStorage.getItem('data');
+if (previousDataJSON !== null) {
+  data.entries.push(JSON.parse(previousDataJSON));
+  data.entries.flat();
+  data.nextEntryId = (JSON.parse(previousDataJSON)).nextEntryId;
```

Comment:
If you just parse the old `data` object and replace the current `data` object with the old one from localStorage, this won't be necessary